cpp/src/codegen/node.h

```diff
@@ -38,6 +40,11 @@ class Node {
   /// Derived classes should simply invoke the Visit api of the visitor.
   virtual Status Accept(NodeVisitor &visitor) const = 0;
 
+  // for debugging purposes
+  virtual std::string to_string(uint32_t indent) = 0;
```

Comment:
- since this is not an accessor, it should be ToString().
- Adding the indent param makes this a non-conventional ToString(). Can the indents be handled in the caller ?